doxygen/contributor_help_pages/common_pitfalls.md

```diff
@@ -190,6 +190,38 @@ The general rules to follow for passing values to a function are:
 2. If you are writing a function for reverse mode, pass values by `const&`
 3. In prim, if you are confident and working with larger types, use perfect forwarding to pass values that can be moved from. Otherwise simply pass values by `const&`.
 
+
+### Using auto is Dangerous With Eigen Matrix Functions in Reverse Mode
+
+Like in [Eigen](https://eigen.tuxfamily.org/dox/TopicPitfalls.html), the use of `auto` with the Stan math library should be used with care. Along with the cautions mentioned in the Eigen docs, there are also memory considerations when using reverse mode automatic differentiation. When returning from a function in the Stan math library with an Eigen matrix output with a scalar `var` type, the actual returned type will often be an `arena_matrix<Eigen::Matrix<...>>`. The `arena_matrix` class is an Eigen matrix where the underlying array of memory is located in Stan's memory arena. The `arena_matrix` that is returned by Stan functions is normally the same one resting in the callback used to calculate gradients in the reverse pass. Directly changing the elements of this matrix would also change the memory the reverse pass callback sees which would result in incorrect calculations.
```

Comment:
Let's reverse the way the first sentence is written. Instead of "Like in Eigen..." I think it'd be better if we got to the punchline first: "The use of `auto` with the Stan Math library should be used with care, like in Eigen..."